src/packs/checker.rs

```diff
@@ -262,9 +263,43 @@ pub(crate) fn update(
     Ok(())
 }
 
+pub(crate) fn remove_unnecessary_dependencies(
+    configuration: &Configuration,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let unnecessary_dependencies = get_unnecessary_dependencies(configuration);
+    for (package_name, dependency_names) in unnecessary_dependencies.iter() {
+        remove_reference_to_dependency(
+            configuration,
+            package_name,
+            dependency_names,
+        );
+    }
+    Ok(())
+}
+
 pub(crate) fn check_unnecessary_dependencies(
     configuration: &Configuration,
 ) -> Result<(), Box<dyn std::error::Error>> {
+    let unnecessary_dependencies = get_unnecessary_dependencies(configuration);
+    if unnecessary_dependencies.is_empty() {
+        Ok(())
+    } else {
+        for (package_name, dependency_names) in unnecessary_dependencies.iter()
+        {
+            for dependency_name in dependency_names {
+                println!(
+                    "{} depends on {} but does not use it",
+                    package_name, dependency_name
+                )
+            }
+        }
+        Err("List unnecessary dependencies failed".into())
+    }
+}
+
+fn get_unnecessary_dependencies(
```

Comment:
not a big deal but – Another option here is to return a hash map of pack to unnecessary dependencies. Then the remove dependencies method has to do less work to get the pack and only has to write the file once per pack.